### PR TITLE
create certificates from class parameter (hiera)

### PR DESCRIPTION
**Pull request by nod0n**
#### Pull Request (PR) description
Support passing a hash of certificates from Hiera to the `letsencrypt` class.
This PR just iterates over the hash and creates `letsencrypt::certonly` for each entry.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -52,30 +53,31 @@
...
+  String $config_file                = "${config_dir}/cli.ini",
+  Hash $config                       = { 'server' => 'https://acme-v02.api.letsencrypt.org/directory' },
+  String $cron_scripts_path          = "${facts['puppet_vardir']}/letsencrypt",
+  String $cron_owner_group           = 'root',
+  Boolean $manage_config             = true,
+  Boolean $manage_install            = true,
+  Boolean $agree_tos                 = true,
+  Boolean $unsafe_registration       = false,
+  Integer[2048] $key_size            = 4096,
+  Hash[String[1],Hash] $certificates = {},
```
Comment (traylenator) [inline manifests/init.pp:71]: Can you add some spec tests for when this parameter is set that it does contain some `letsencrypt::certonly` entries.

Comment (nod0n) [reply manifests/init.pp:71]: I'm very new to rspec but: https://github.com/voxpupuli/puppet-letsencrypt/pull/271/commits/9723ce55ffb1410b79b0f0ae4c7776fb53b14ece

File: spec/classes/letsencrypt_spec.rb [hunk 2]
```diff
@@ -141,6 +141,13 @@
           it { is_expected.to contain_file('/foo/bar/baz').with(ensure: 'directory') }
         end
 
+        describe 'with list of certificates' do
+          let(:additional_params) { { certificates: { 'foo' => {}, 'a' => {} } } }
```
Comment (traylenator) [inline spec/classes/letsencrypt_spec.rb:145]: Pass in some random parameters to `foo` for instance just to check it does what is expected.

**Comment by nod0n** [comment]
rubocop is happy now. :)
should I squash the commits? Any other suggestions?

**Pull request merged**